### PR TITLE
[FEMR179] append dates to research drop down

### PR DESCRIPTION
**Pull request by kevinzurek**
Takes care of:
https://teamfemr.atlassian.net/browse/FEMR-179

Also fixes some bad routes that were called for medication typeahead & location of typeahead library.

Also disables the Filter Date option when selecting a Filter Trip from the drop down

### CODE REVIEW

File: app/femr/util/calculations/dateUtils.java [hunk 1]
```diff
@@ -160,12 +160,41 @@ public static String getFriendlyDate(DateTime dateTime){
...
     }
 
+    /**
+     * Takes a date object and formats it into a user friendly string
+     *
+     * Format: (MM/dd/yyyy)
+     * Format: (08/02/1989)
+     *
+     * @param date the date you would like to format, not null
+     * @return the sexy looking date string or null if errors
```
Comment (AJSclyn) [inline app/femr/util/calculations/dateUtils.java]: 👍

**Comment by AJSclyn** [comment]
Looks good from my end.

**Comment by kevinzurek** [comment]
I dare you to merge it :)

**Comment by AJSclyn** [comment]
Merged manually

https://github.com/FEMR/femr/commit/5f91a8791bdec55bcafaae889ff0ce063324f78c

**Pull request closed without merging**